pex/resolve/resolver_configuration.py

```diff
@@ -31,6 +31,26 @@ class ResolverVersion(Enum["ResolverVersion.Value"]):
     class Value(Enum.Value):
         pass
 
+    @staticmethod
+    def _supports_legacy_resolver(pip_version=None):
+        # type: (Optional[PipVersionValue]) -> bool
+        pip_ver = pip_version or PipVersion.DEFAULT
+        return pip_ver.version < Version("23.2")
```

Comment:
It looks like the underlying issue will get fixed before 23.2 is released (https://github.com/pypa/pip/pull/12140). I'll probably follow up, leaving all the code in this change in place but changing this function to just return `True` for all cases so that `< Version("X.Y.Z")` can go back in when Pip does actually drop support for the legacy resolver.